lib/rack/error_handler.rb

```diff
@@ -0,0 +1,23 @@
+module Rack
+  class ErrorHandler
+    def initialize(app)
+      @app = app
+    end
+
+    def bad_request(error)
+      [400, {}, ["Bad Request: #{error.message}"]]
+    end
+
+    def internal_server_error(error)
+      [500, {}, ["Internal Server Error: #{error.message}"]]
+    end
+
+    def call(env)
+      @app.call(env)
+    rescue BadRequest => error
+      return bad_request(error)
+    rescue => error
```

Comment:
May be worth rescuing both ScriptError and StandardError here.

Reply:
What status codes would we map them to?

Reply:
Okay, all mapping to internal server error.